app/views/handlers/individual_response.py

```diff
@@ -137,22 +139,44 @@ def handle_get(self):
             self.individual_section_id
         )
 
-        if self._list_item_id:
+        if self._is_hub_journey:
+            previous_location_url = url_for("questionnaire.get_questionnaire")
+        else:
             previous_location_url = url_for(
                 "questionnaire.block",
                 list_name=self._list_name,
                 list_item_id=self._list_item_id,
                 block_id=individual_section_first_block_id,
             )
-        else:
-            previous_location_url = url_for("questionnaire.get_questionnaire")
 
         return render_template(
             template="individual_response/interstitial",
             language=self._language,
             previous_location_url=previous_location_url,
         )
 
+    def handle_post(self):
+        if self._list_item_id:
+            return redirect(
+                url_for(".get_individual_response_how", list_item_id=self._list_item_id)
+            )
+
+        self._list_name = self._schema.get_individual_response_list()
+        list_model = self._questionnaire_store.list_store[self._list_name]
+        non_primary_members = []
+
+        for list_item in list_model.items:
+            if list_item != list_model.primary_person:
+                non_primary_members.append(list_item)
+
+        if len(non_primary_members) == 1:
```

Comment:
I think we should have a property on list model to return the non primary list item ids. Which can then be reused here, as well as in:
- https://github.com/ONSdigital/eq-questionnaire-runner/pull/238/files#diff-1b7aa7bc01cf9df6f320a4fd4a937b8aR544-R545
- https://github.com/ONSdigital/eq-questionnaire-runner/pull/238/files#diff-c8d1a13e851293923c3a04e3f7d0d1c0R186-R188